source/agora/consensus/data/Block.d

```diff
@@ -843,3 +847,55 @@ unittest
     assert(merkle_path[3] == habcdefgh);
     assert(block.header.merkle_root == Block.checkMerklePath(hi, merkle_path, 8));
 }
+
+/// demonstrate signing two blocks at height 1 to reveal private node key
+unittest
+{
+    import agora.consensus.data.genesis.Test: GenesisBlock;
+    import agora.crypto.ECC: Scalar, Point;
+    import agora.crypto.Schnorr;
+    import agora.utils.Test;
+    import std.format;
+
+    Hash random_seed1 = "seed1".hashFull();
+    Hash random_seed2 = "seed2".hashFull();
+    Hash preimage = "preimage".hashFull();
+
+    // Generate two blocks at height 1
+    auto block1 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(2).map!(txb => txb.sign()).array(), 10, random_seed1, 6);
+    auto block2 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(3).map!(txb => txb.sign()).array(), 10, random_seed2, 6);
+
+    Scalar v = genesis_validator_keys[0].secret;
+    assert(v.isValid(), "v is not a valid Scalar!");
+    Point V = v.toPoint();
+    const Scalar rc = Scalar(hashMulti(v, "consensus.signature.noise", 0));
+    const Scalar r = rc + Scalar(preimage);
+    const Point R = r.toPoint();
+
+    // Two messages
+    Scalar c1 = block1.hashFull();
+    Scalar c2 = block2.hashFull();
+    assert(c1 != c2);
+
+    // Sign with same r twice
+    Signature sig1 = sign(v, R, r, c1);
+    Signature sig2 = sign(v, R, r, c2);
```

Comment:
Why is this not using `createBlockSignature` ?

Reply:
Yes good idea. Updated.